src/server/auth/server/api_server.go

```diff
@@ -124,35 +130,122 @@ func NewAuthServer(env Env, public, requireNoncriticalServers, watchesEnabled bo
-					return errors.EnsureStack(err)
+		}
+		// handle oidc clients & this cluster's auth config
+		if a.env.Config.AuthConfig != "" && a.env.Config.IdentityClients != "" {
+			var config auth.OIDCConfig
+			var clients []identity.OIDCClient
+			if err := yaml.Unmarshal([]byte(a.env.Config.AuthConfig), &config); err != nil {
+				return errors.Wrapf(err, "unmarshal auth config: %q", a.env.Config.AuthConfig)
+			}
+			if err := yaml.Unmarshal([]byte(a.env.Config.IdentityClients), &clients); err != nil {
+				return errors.Wrapf(err, "unmarshal identity clients: %q", a.env.Config.IdentityClients)
+			}
+			for _, c := range clients {
+				if c.Id == config.ClientID {
+					c.Secret = config.ClientSecret
+				}
+				if c.Id == a.env.Config.ConsoleOAuthID {
+					c.Secret = a.env.Config.ConsoleOAuthSecret
+				}
+				if !a.env.Config.EnterpriseMember {
+					if _, err := a.env.GetIdentityServer().CreateOIDCClient(ctx, &identity.CreateOIDCClientRequest{Client: &c}); err != nil {
+						if !identity.IsErrAlreadyExists(err) {
+							return errors.Wrapf(err, "create oidc client %q", c.Name)
+						}
+						if _, err := a.env.GetIdentityServer().UpdateOIDCClient(ctx, &identity.UpdateOIDCClientRequest{Client: &c}); err != nil {
+							return errors.Wrapf(err, "update oidc client %q", c.Name)
+						}
+					}
+				} else {
+					ec, err := client.NewFromURI(a.env.Config.EnterpriseServerAddress)
+					if err != nil {
+						return errors.Wrapf(err, "connect to enterprise server")
+					}
+					ec.SetAuthToken(a.env.Config.EnterpriseServerToken)
+					if _, err = ec.IdentityAPIClient.CreateOIDCClient(ec.Ctx(), &identity.CreateOIDCClientRequest{Client: &c}); err != nil {
+						if !identity.IsErrAlreadyExists(err) {
+							return errors.Wrapf(err, "create oidc client %q", c.Name)
+						}
+						if _, err := ec.IdentityAPIClient.UpdateOIDCClient(ctx, &identity.UpdateOIDCClientRequest{Client: &c}); err != nil {
+							return errors.Wrapf(err, "update oidc client %q", c.Name)
+						}
+					}
+
+				}
+			}
+			if _, err := a.SetConfiguration(ctx, &auth.SetConfigurationRequest{Configuration: &config}); err != nil {
+				return err
+			}
+		}
+		// cluster role bindings
+		if a.env.Config.AuthClusterRoleBindings != "" {
+			var roleBinding map[string][]string
+			if err := yaml.Unmarshal([]byte(a.env.Config.AuthClusterRoleBindings), &roleBinding); err != nil {
+				return errors.Wrapf(err, "unmarshal auth cluster role bindings: %q", a.env.Config.AuthClusterRoleBindings)
+			}
+			existing, err := a.GetRoleBinding(ctx, &auth.GetRoleBindingRequest{
+				Resource: &auth.Resource{Type: auth.ResourceType_CLUSTER},
+			})
+			if err != nil {
+				return errors.Wrapf(err, "get cluster role bindings")
+			}
+			for p := range existing.Binding.Entries {
+				// `pach:` user role bindings cannot be modified
+				if strings.HasPrefix(p, auth.PachPrefix) || strings.HasPrefix(p, auth.InternalPrefix) {
+					continue
```

Comment:
Let's log something here, so users don't have to guess as to why this was skipped.  I could also be talked into crashing (config is broken, try again), but I'm fine with just a log.